### ISSUE TITLE
getReadAlignments method required

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#247]
The `datamodel.reads.ReadGroup` class needs a method called `getReadAlignments`. This should be directly analogous to the  `datamodel.variants.VariantSet.getVariants` method, and map directly to the `SearchReadsRequest` protocol method. Specifically, we should have
```python
getReadAlignments(self, referenceName, referenceId, start, end):
     # Get the corresponding reads from PySAM, convert to GA4GH objects and yield each one in turn 
     # to the calling function.
```

This should be tested in `datadriven/test_reads.py`. We should test for searching over all reads, and subsets of the reads within the file. Ideally, we would like to use a different library to provide an independent code path, but we might have to use pysam for the tests also. (For the VCF testing, we use pysam as our production code and pyvcf as an independent test, making sure that (a) we're not reusing the same code in our production code and test code; and (b) having an independent sanity check on our usage of pysam.)

**Comment by macieksmuga** [comment]
@jeromekelleher sorry for asking this late, but why both the `referenceName` and `referenceId` arguments? From the implementation, it looks like at most one can be passed in, otherwise an exception is raised. This seems like a strange convention to me (not followed anywhere else in the codebase, as far as I can tell). Since `referenceName` can be computed internally from `referenceId`, is there some intended use of this method that would necessitate passing in the former as opposed to latter argument?

**Comment by jeromekelleher** [comment]
Don't ask me @macieksmuga, I have no idea! This is coming from the schemas, which were written before I started working on the project.

**Comment by pgrosu** [comment]
Maciek, Jerome - it already got removed in the following PR this past December, and only `referenceId` now remains:

https://github.com/ga4gh/schemas/pull/193

~p

**Comment by macieksmuga** [comment]
thanks @pgrosu! The discussion in that PR provides the answer I was looking for. Will comment with this on the PR to be merged here...

**Comment by pgrosu** [comment]
Gladly @macieksmuga - we're all helping each other :)

**Comment by dcolligan** [comment]
Fixed in #249
